addons/mail/static/src/messaging/entity/thread/thread.js

```diff
@@ -14,105 +16,102 @@ function ThreadFactory({ Entity }) {
-            return this.allChannels.filter(channel => channel.isPinned);
-        }
+         * @param {Object} data
+         * @return {Object}
+         */
+        static convertData(data) {
+            const data2 = {};
+            if ('channel_type' in data) {
+                data2.channel_type = data.channel_type;
+                data2.model = 'mail.channel';
+            }
+            if ('correspondent_name' in data) {
+                data2.correspondent_name = data.correspondent_name;
+            }
+            if ('create_uid' in data) {
+                data2.create_uid = data.create_uid;
+            }
+            if ('custom_channel_name' in data) {
+                data2.custom_channel_name = data.custom_channel_name;
+            }
+            if ('group_based_subscription' in data) {
+                data2.group_based_subscription = data.group_based_subscription;
+            }
+            if ('id' in data) {
+                data2.id = data.id;
+            }
+            if ('is_minimized' in data && 'state' in data) {
+                data2.foldState = data.is_minimized ? data.state : 'closed';
+            }
+            if ('is_moderator' in data) {
+                data2.is_moderator = data.is_moderator;
+            }
+            if ('mass_mailing' in data) {
+                data2.mass_mailing = data.mass_mailing;
+            }
+            if ('moderation' in data) {
+                data2.moderation = data.moderation;
+            }
+            if ('message_needaction_counter' in data) {
+                data2.message_needaction_counter = data.message_needaction_counter;
+            }
+            if ('message_unread_counter' in data) {
+                data2.message_unread_counter = data.message_unread_counter;
+            }
+            if ('name' in data) {
+                data2.name = data.name;
+            }
+            if ('public' in data) {
+                data2.public = data.public;
+            }
+            if ('seen_message_id' in data) {
+                data2.seen_message_id = data.seen_message_id;
+            }
+            if ('seen_partners_info' in data) {
+                data2.seen_partners_info = data.seen_partners_info;
+            }
+            if ('uuid' in data) {
+                data2.uuid = data.uuid;
+            }
 
-        /**
-         * @static
-         * @returns {mail.messaging.entity.Thread[]}
-         */
-        static get allUnreadChannels() {
-            return this.allChannels.filter(
-                channel => channel.message_unread_counter > 0
-            );
-        }
+            // relation
+            if ('direct_partner' in data) {
+                if (!data.direct_partner) {
+                    data2.directPartner = [['unlink-all']];
+                } else {
+                    data2.directPartner = [
+                        ['insert', this.env.entities.Partner.convertData(data.direct_partner[0])]
```

Comment:
Not sure what is `direct_partner[0]` (id?) but shouldn't `convertData` take object?